File: tests/fixtures/sample_codebase/go/sample.go
// Go sample code

package main

type User struct {
    Name string
    ID   int
}

type Service interface {
    GetUser(id int) (*User, error)
}

type UserService struct{}

func (s *UserService) GetUser(id int) (*User, error) {
    return &User{Name: "test", ID: id}, nil
}

func CalculateSum(a, b int) int {
    return a + b
}

File: tests/fixtures/sample_codebase/rust/sample.rs
// Rust sample code

struct User {
    name: String,
    id: u32,
}

impl User {
    fn new(name: String, id: u32) -> Self {
        User { name, id }
    }

    fn get_name(&self) -> &str {
        &self.name
    }
}

enum Status {
    Active,
    Inactive,
}

fn calculate_sum(a: i32, b: i32) -> i32 {
    a + b
}